PACKAGE = "l3cam_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Parameters
# gen.add(name, paramtype, level, description, default, min, max, edit_method)

# CHANGE_NETWORK_CONFIGURATION
gen.add("change_network_configuration_ip_address", str_t, 0, "Change IP address of network configuration", "192.168.5.15")
gen.add("change_network_configuration_netmask", str_t, 1, "Change netmask of network configuration", "255.255.255.0")
gen.add("change_network_configuration_gateway", str_t, 2, "Change gateway of network configuration", "0.0.0.0")
gen.add("enable_network_configuration_dhcp", bool_t, 3, "Enable DHCP of network configuration", False)

exit(gen.generate(PACKAGE, "l3cam_ros", "General"))
